src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,634 @@
+		$activate_url = wp_nonce_url( 'plugins.php?action=activate&amp;plugin=' . rawurlencode( $dependency['file'] ) . '&amp;plugin_status=all', 'activate-plugin_' . $dependency['file'] );
+
+		$this->notices[] = array(
+			'content' => sprintf(
+				/* translators: %1$s: The plugin we want to activate. %2$s: The name of the plugin to install. %3$s: "Activate" button. */
+				__( 'Plugin "%1$s" depends on plugin "%2$s" to be activated. %3$s' ),
+				$plugin['Name'],
+				$dependency['name'],
+				'<a href="' . $activate_url . '">' . __( 'Activate plugin' ) . '</a>'
+			),
+		);
+	}
+
+	/**
+	 * Get an array of plugins that should be activated but are not,
+	 * due to missing/unmet dependencies.
+	 *
+	 * @since 5.9.0
+	 * @access public
+	 *
+	 * @return array
+	 */
+	public function get_plugins_to_activate() {
+		return get_option( $this->pending_plugin_activations_option, array() );
+	}
+
+	/**
+	 * Set plugin to the to-be-activated queue.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @param string $plugin The plugin file.
+	 *
+	 * @return bool
+	 */
+	protected function add_plugin_to_queue( $plugin ) {
+		$queue = $this->get_plugins_to_activate();
+		if ( in_array( $plugin, $queue, true ) ) {
+			return true;
+		}
+		$queue[] = $plugin;
+		return update_option( $this->pending_plugin_activations_option, $queue );
+	}
+
+	/**
+	 * Remove plugin from the to-be-activated queue.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @param string $plugin The plugin file.
+	 *
+	 * @return bool
+	 */
+	protected function remove_plugin_from_queue( $plugin ) {
+		$queue = $this->get_plugins_to_activate();
+		if ( ! in_array( $plugin, $queue, true ) ) {
+			return true;
+		}
+		return update_option( $this->pending_plugin_activations_option, array_diff( $queue, array( $plugin ) ) );
+	}
+
+	/**
+	 * Check if a plugin is part of a circular dependencies loop.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @param string $plugin_file The plugin file.
+	 * @param array  $previous    If this is a dependency of a dependency,
+	 *                            this array contains all previous levels of dependencies.
+	 *
+	 * @return bool
+	 */
+	protected function in_circular_dependency( $plugin_file, $previous = array() ) {
+		if ( isset( $this->circular_dependencies[ $plugin_file ] ) ) {
+			return $this->circular_dependencies[ $plugin_file ];
+		}
+
+		if ( in_array( $plugin_file, $previous, true ) ) {
+			$this->circular_dependencies[ $plugin_file ] = true;
+		}
+
+		$plugin_dependencies = $this->get_plugin_dependencies( $plugin_file );
+
+		foreach ( $plugin_dependencies as $dependency ) {
+			$dependency_file = $this->get_plugin_file_from_slug( $dependency['slug'] );
+			if ( $this->in_circular_dependency( $dependency_file, array_merge( $previous, array( $plugin_file ) ) ) ) {
+				$this->circular_dependencies[ $plugin_file ] = true;
+				$this->circular_dependencies[ $dependency_file ] = true;
```

Comment:
WPCS violation here.  The assignment operators should be in the same column position :-)